bauble/src/builtin.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{
+    Bauble, BaubleAllocator, ToRustErrorKind, Val, Value,
+    path::TypePath,
+    types::{Type, TypeRegistry},
+};
+use std::{cell::UnsafeCell, fmt::Debug, marker::PhantomData};
+
+/// The builtin reference type.
+///
+/// Corresponds to a reference to `T` in Bauble.
+///
+/// `S` is the inner representation used for `TypePath`.
+pub struct Ref<T, S = String> {
```

Comment:
I feel like this type should live in the tests, until a use case outside of testing exists.